### PR TITLE
Add code analysis using the lint and vet tools to github->actions->wo…

### PR DESCRIPTION
**Pull request by tdrozdovsky**
…rkflow

Signed-off-by: Taras Drozdovskyi <t.drozdovsky@samsung.com>

# Description

Added code analysis with additional utilities: `goling` & `vet` to `github->Actions->workflow`. This is one more step to improve our code and a step to increase the CII Best Practice rate.
**Note that the `golint` utility issues `ERROR` messages - they are not critical** and we will gradually fix them (most of them require adding a description of functions, etc.) 

## Type of change

Please delete options that are not relevant.

- [x] Github->Actions->workflow update

# How Has This Been Tested?

You can see in Action tab - build workflow result for x86_64 arch.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .github/workflows/edge-home-orchestration-go.yml [hunk 1]
```diff
@@ -38,6 +39,10 @@ jobs:
       - name: Build the container
         run: ./build.sh container ${{ matrix.arch }}
 
+      - name: Analysis using the lint & vet tools
+        if:  ${{ matrix.arch == 'x86_64' }}
```
Comment (t25kim) [inline .github/workflows/edge-home-orchestration-go.yml:43]: It is not necessary to run the analysis on every OS.

**Review by t25kim — COMMENTED** [review]
Thank you for the PR!
How about splitting this analysis to `lint` job rather than within `build` job?

File: .github/workflows/edge-home-orchestration-go.yml (hunk 1 above)
Comment (tdrozdovsky) [reply .github/workflows/edge-home-orchestration-go.yml:43]: I could not give preference to any particular system, so I'm testing on all! )

**Comment by tdrozdovsky** [comment]
> Thank you for the PR!
> How about splitting this analysis to `lint` job rather than within `build` job?

What reason do you see for this? if it is I'll do it!

**Comment by t25kim** [comment]
> > Thank you for the PR!
> > How about splitting this analysis to `lint` job rather than within `build` job?
> 
> What reason do you see for this? if it is I'll do it!

I'd like to go with `microservice` concept rather than `monolithic`.
`lint` is not a critical job to run edge-orchestration but it takes time to inspect code.

File: .github/workflows/edge-home-orchestration-go.yml (hunk 1 above)
Comment (t25kim) [reply .github/workflows/edge-home-orchestration-go.yml:43]: > I could not give preference to any particular system, so I'm testing on all! )

Do with latest version! 😁

**Comment by tdrozdovsky** [comment]
@t25kim 
I made the changes you wrote above.

Please note how the action table has changed after your comments 

Before:
| action\arch                      | x86_64 (16.04) | x86_64 (18.04) | x86_64 (20.04) | x86 | arm64 | arm | 
|---------------------------|----------------|-----------------|----------------|-----|--------|------|
| ./build.sh                         | x                      |                          |                        |        |            |        |
| ./build.sh container [arch]| x                      | x                        | x                      | x      | x         | x      |
| Test Suite                        | x                      | x                        | x                     |        |            |         |
|  Test using lint & wet      | x                      | x                        | x                     |        |            |         |
| ./build.sh object (TBD)     |                         |                           |                       |        |            |         |

After:
| action\arch                      | x86_64 (16.04) | x86_64 (18.04) | x86_64 (20.04) | x86 | arm64 | arm | 
|---------------------------|----------------|-----------------|----------------|-----|--------|------|
| ./build.sh                         | x                      |                          |                        |        |            |        |
| ./build.sh container [arch]| x                      | x                        | x                      | x      | x         | x      |
| Test Suite                        |                        |                           | x                     |        |            |         |
| Test using lint & wet       |                        |                           | x                     |        |            |         |
| ./build.sh object (TBD)     |                         |                           |                       |        |            |         |

I understand that the result of tests will not change in different versions of Ubuntu OS, but in this case, you can be sure that the tests run successfully or not. 

**I'm also not sure if this will significantly speed up the process, since builds and tests for different version OSes are run in parallel.**

**Review by MoonkiHong — APPROVED** [review]
LGTM.

**Comment by t25kim** [comment]
> @t25kim
> I made the changes you wrote above.
> 
> Please note how the action table has changed after your comments
> 
> Before:
> 
> action\arch	x86_64 (16.04)	x86_64 (18.04)	x86_64 (20.04)	x86	arm64	arm
> ./build.sh	x					
> ./build.sh container [arch]	x	x	x	x	x	x
> Test Suite	x	x	x			
> Test using lint & wet	x	x	x			
> ./build.sh object (TBD)						
> After:
> 
> action\arch	x86_64 (16.04)	x86_64 (18.04)	x86_64 (20.04)	x86	arm64	arm
> ./build.sh	x					
> ./build.sh container [arch]	x	x	x	x	x	x
> Test Suite			x			
> Test using lint & wet			x			
> ./build.sh object (TBD)						
> I understand that the result of tests will not change in different versions of Ubuntu OS, but in this case, you can be sure that the tests run successfully or not.
> 
> **I'm also not sure if this will significantly speed up the process, since builds and tests for different version OSes are run in parallel.**

There's a miscommunication. It was not intended to remove the TC running in other OS scenarios. 
Please find my [PR](https://github.com/tdrozdovsky/edge-home-orchestration-go/pull/1) on your repo and hope you to understand my `microservice` concept.

**Comment by tdrozdovsky** [comment]
@t25kim Now I understand what you mean!
In this case, it might make sense to create two new workflow files: 1st - "Lint and Vet Analysis" and 2nd "Test Suite". I suggest doing this in order not to create a very large `edge-home-orchestration-go.yml` file (as I hope we will soon add a build for the native and android versions). **What do you think about it?**

**Comment by t25kim** [comment]
> @t25kim Now I understand what you mean!
> In this case, it might make sense to create two new workflow files: 1st - "Lint and Vet Analysis" and 2nd "Test Suite". I suggest doing this in order not to create a very large `edge-home-orchestration-go.yml` file (as I hope we will soon add a build for the native and android versions). **What do you think about it?**

Your idea is good! Thanks.

File: .github/workflows/lint -vet-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+---
```
Comment (t25kim) [inline .github/workflows/lint -vet-analysis.yml:1]: Please remove this duplicated file.

File: .github/workflows/test-suite.yml [hunk 3]
```diff
@@ -19,27 +18,16 @@ jobs:
...
           echo "$HOME/go/bin" >> $GITHUB_PATH
           sudo mkdir -p /var/edge-orchestration/mnedc
           echo -e '192.168.0.125\n3334' | sudo tee /var/edge-orchestration/mnedc/client.config
-      - name: Install Qemu
-        if:  ${{ matrix.arch != 'x86_64' }}    
-        run: |        
-          sudo apt-get update
-          sudo apt-get install -y qemu binfmt-support qemu-user-static
   
       - name: Build the project
```
Comment (t25kim) [inline .github/workflows/test-suite.yml:27]: `go mod vendor` is enough.

File: .github/workflows/test-suite.yml [hunk 4]
```diff
@@ -19,27 +18,16 @@ jobs:
...
-        if:  ${{ matrix.arch == 'x86_64' }}    
         run: ./build.sh
```
Comment (t25kim) [inline .github/workflows/test-suite.yml:28]:
```suggestion
        run: make go-vendor
```

File: .github/workflows/test-suite.yml [hunk 5]
```diff
@@ -19,27 +18,16 @@ jobs:
...
-        if:  ${{ matrix.arch == 'x86_64' }}    
         run: ./build.sh 
 
-      - name: Build the container
-        run: ./build.sh container ${{ matrix.arch }}
-
       - name: Run the Test Suite
-        if:  ${{ matrix.arch == 'x86_64' }}
         run: |
           docker stop edge-orchestration || true
```
Comment (t25kim) [inline .github/workflows/test-suite.yml:32]: If we don't build, this line should be deleted.

**Review by t25kim — COMMENTED** [review]
Thank you for your contribution!
Please take a look my comments.

**Review by t25kim — APPROVED** [review]
LGTM. Thanks!

**Review by MoonkiHong — APPROVED** [review]
LGTM.

**Pull request merged**